api/v1/reviewers/recommendations/ReviewerRecommendationController.php

```diff
@@ -0,0 +1,219 @@
+            ->name('reviewer.recommendations.edit.status')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::delete('{reviewerRecommendationId}', $this->delete(...))
+            ->name('reviewer.recommendations.delete')
+            ->whereNumber(['reviewerRecommendationId']);
+    }
+
+    /**
+     * Get specific recommendation response
+     */
+    public function get(Request $illuminateRequest): JsonResponse
+    {
+        $recommendation = ReviewerRecommendation::find($illuminateRequest->route('reviewerRecommendationId'));
+
+        return response()->json(
+            (new ReviewerRecommendationResource($recommendation))->toArray($illuminateRequest),
+            Response::HTTP_OK
+        );
+    }
+
+    /**
+     * Get all recommendations response
+     */
+    public function getMany(Request $illuminateRequest): JsonResponse
+    {
+        $recommendations = ReviewerRecommendation::query()
+            ->withContextId(Application::get()->getRequest()->getContext()->getId())
+            ->get();
+
+        return response()->json([
+            'items' => ReviewerRecommendationResource::collection($recommendations),
+            'itemMax' => $recommendations->count(),
+        ], Response::HTTP_OK);
+    }
+
+    /**
+     * Add new recommendation
+     */
+    public function add(AddReviewerRecommendation $illuminateRequest): JsonResponse
+    {
+        $validateds = $illuminateRequest->validated();
+
+        $recommendation = ReviewerRecommendation::create($validateds);
+
+        return response()->json(
+            (new ReviewerRecommendationResource($recommendation->refresh()))
+                ->toArray($illuminateRequest),
+            Response::HTTP_OK
+        );
+    }
+
+    /**
+     * Update existing recommendation
+     */
+    public function edit(EditReviewerRecommendation $illuminateRequest): JsonResponse
+    {
+        $validated = $illuminateRequest->validated();
+
+        $recommendation = ReviewerRecommendation::find($illuminateRequest->route('reviewerRecommendationId'));
+
+        if (!$recommendation->removable) {
+            return response()->json([
+                'error' => __('api.406.notAcceptable'),
+            ], Response::HTTP_NOT_ACCEPTABLE);
+        }
+
+        if (!$recommendation->update($validated)) {
+            return response()->json([
+                'error' => __('api.409.resourceActionConflict'),
+            ], Response::HTTP_CONFLICT);
+        }
+
+        return response()->json(
+            (new ReviewerRecommendationResource($recommendation->refresh()))
+                ->toArray($illuminateRequest),
+            Response::HTTP_OK
+        );
+    }
+
+    /**
+     * Update the status of existing recommendation
+     */
+    public function updateStatus(UpdateStatusReviewerRecommendation $illuminateRequest): JsonResponse
+    {
+        $validated = $illuminateRequest->validated();
```

Comment:
Shouldn't this return an error if not validated?

Reply:
As described at https://github.com/pkp/ojs/pull/4505#discussion_r2041123053 , if validation failed, it will be handled in the `UpdateStatusReviewerRecommendation` with proper response before reaching here . Also the `validated` does not run the validation but get only the validated data/input .